### PR TITLE
Fix #53: Fix issue with running tests

### LINKED ISSUE #53: Issue with running tests.
I investigate the test failures and I was able to fix some of them. Others are failing for legitimate reasons, which is why I created this issue.

1. The scheduler gets created at https://github.com/oppia/oppiabot/blob/6d723a95549353bb1641325f62efc95c8f420f64/index.js#L11, but in the tests, the app (oppiabot) is initialized https://github.com/oppia/oppiabot/blob/6d723a95549353bb1641325f62efc95c8f420f64/spec/oppiabotSpec.js#L16 before `robot.auth` gets mocked https://github.com/oppia/oppiabot/blob/6d723a95549353bb1641325f62efc95c8f420f64/spec/oppiabotSpec.js#L82 and the scheduler calls `robot.auth` https://github.com/probot/scheduler/blob/8ad237c69116894a5579d360e544f2586191fa7a/index.js#L77 before it is mocked so it calls the actual `robot.auth` and not the mocked implementation.

A possible fix for this would be to mock `robot.auth` before initializing oppiabot. I did that and it fixed the initial problem but provided another problem: the mock github data which is being resolved misses some data required by the scheduler.
2. Before every event gets dispatched, we check if the repo owner is allowed to run the bot https://github.com/oppia/oppiabot/blob/6d723a95549353bb1641325f62efc95c8f420f64/index.js#L23. Going through the mock payload, the repo owner is set to oppia, so adding oppia to the whitelisted account in the .env will make this work.

After fixing the above issues, 8 tests pass and 6 fail. 
3. This test https://github.com/oppia/oppiabot/blob/6d723a95549353bb1641325f62efc95c8f420f64/spec/oppiabotSpec.js#L103 checks that the function is called with 3 arguments, but the function is only called with one (context) which is expected since probot calls the handlers with just one argument. 
4. https://github.com/oppia/oppiabot/blob/6d723a95549353bb1641325f62efc95c8f420f64/spec/oppiabotSpec.js#L108 tests that the first argument passed is the username of the user, but it is instead the context , and again this is the expected behaviour.
Other tests fail for similar reason. maybe this is from an earlier implementation?

### PR DESCRIPTION
**Pull request by jameesjohn**
## Explanation
Fixes #53: Fixes issue with running tests by:
1. Exporting a module which has the create-scheduler function (probot-scheduler) so it can be mocked.
2. Modify existing tests.

## Checklist
- [x] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [x] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

### CODE REVIEW

**Comment by jameesjohn** [comment]
PTAL @ankita240796, @kevinlee12. Thanks

File: lib/scheduler.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const createScheduler = require('probot-scheduler');
+
+module.exports = { createScheduler };
```
Comment (kevinlee12) [inline lib/scheduler.js:3]: Add newline after this.

Comment (jameesjohn) [reply lib/scheduler.js]: Done

**Comment by jameesjohn** [comment]
Closing this since a new PR will be created to fix this in the updated probot.

**Pull request closed without merging**